src/Stoa.ts

```diff
@@ -1704,6 +1711,136 @@ class Stoa extends WebService
                 res.send(500).send("Failed to data lookup")
             })
     }
+    /**
+     *  Stoa emits the latest Boa stats using sockets on new block recieved. 
+     * @returns Returns the Promise. If it is finished successfully the `.then`
+     * of the returned Promise is called
+     * and if an error occurs the `.catch` is called with an error.
+     */
+    public emitBoaStats(): Promise<IBOAStats> {
+        return new Promise<IBOAStats>(async (resolve, reject) => {
+               this.ledger_storage.getBOAStats()
+        .then((data: any[]) => {
+            if (!data[0]) {
+                logger.info("Failed to latest BOA stats");
+                return;
+              }
+            else {
+                let boaStats: IBOAStats = {
+                    height: data[0].height,
+                    transactions: data[0].transactions,
+                    validators: data[0].validators,
+                    frozen_coin: 5283595, //FIX ME static data because of unavailability of real data
+                    circulating_supply: 5283535,
+                    active_validators: 155055
+                };
+            this.socket.io.emit(events.server.lateststats, boaStats)
+            logger.info(`Emited Updated BOA stats:  ${boaStats}`)
+            return resolve(boaStats);
+            }
+        })
+        .catch((err) => {
+            logger.error("Failed to latest BOA stats: " + err);
+           return;
+        });
+        });
+    }
+
+    /**
+     *  Stoa emits the updates using sockets on new block recieved
+     * @param height The height of block to emit  
+     * @returns Returns the Promise. If it is finished successfully the `.then`
+     * of the returned Promise is called
+     * and if an error occurs the `.catch` is called with an error.
+     */
+    public emitBlock(height: Height): Promise<boolean> {
```

Comment:
It would be better to emit the data with the interface as below.
``` TypeScript
export interface IEmitBlock
{
    /**
     * block height
     */
    height: string;

    /**
     * hash of block
     */
    hash: string;

    /**
     * timestamp of the block
     */
    time_stamp: string;

    block: Block;
}
```

The interface member `block` can use the data received from Agora as it is.
It would be better to emit it immediately without querying it in the database.

Reply:
We will also need `emitBlockHeader`. The interface to be used for it is as follows.

``` TypeScript

export interface IEmitBlockHeader
{
    /**
     * block height
     */
    height: string;

    /**
     * hash of block
     */
    hash: string;

    /**
     * timestamp of the block
     */
    time_stamp: string;

    block_header: BlockHeader;
}
```
The interface member `block_header` can use the data received from Agora as it is.
It would be better to emit it immediately without querying it in the database.